engine/src/flutter/impeller/renderer/backend/vulkan/render_pass_builder_vk.cc

```diff
@@ -31,7 +41,8 @@ RenderPassBuilderVK& RenderPassBuilderVK::SetColorAttachment(
     SampleCount sample_count,
     LoadAction load_action,
     StoreAction store_action,
-    vk::ImageLayout current_layout) {
+    vk::ImageLayout current_layout,
+    bool is_swapchain) {
```

Comment:
Instead of passing in a bool, why not pass in `vk::ImageLayout final_layout`?

Reply:
We still need the logic in the render_pass_builder because the MSAA texture and resolve always need separate layouts. I'd rather leave this determination in the render_pass_builder